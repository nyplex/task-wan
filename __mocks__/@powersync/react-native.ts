module.exports = {
  PowerSyncDatabase: jest.fn().mockImplementation(() => ({
    connect: jest.fn(),
    ready: true,
  })),
};
