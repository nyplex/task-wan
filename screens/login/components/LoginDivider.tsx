import { HStack } from "@/components/gluestack/hstack";
import Divider from "@/components/layout/Divider";

const LoginDivider = () => {
  return (
    <HStack
      className="items-center mt-8 px-8"
      testID="login-divider">
      <Divider title="Or login with" />
    </HStack>
  );
};

export default LoginDivider;
