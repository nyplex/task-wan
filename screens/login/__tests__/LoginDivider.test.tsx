import React from "react";
import { render } from "@testing-library/react-native";
import LoginDivider from "../components/LoginDivider";

describe("LoginDivider", () => {
  it("renders correctly and matches snapshot", () => {
    const { toJSON, getByText } = render(<LoginDivider />);
    expect(getByText("Or login with")).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders Divider inside HStack", () => {
    const { getByText, getByTestId } = render(<LoginDivider />);
    const divider = getByText("Or login with");
    const parent = getByTestId("login-divider");
    expect(divider).toBeTruthy();
    expect(parent.props.className).toContain("items-center");
  });
});
