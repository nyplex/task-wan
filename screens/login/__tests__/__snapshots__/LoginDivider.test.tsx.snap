// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginDivider renders correctly and matches snapshot 1`] = `
<View
  className="flex-row items-center mt-8 px-8"
  testID="login-divider"
>
  <View
    style={
      {
        "width": "100%",
      }
    }
    testID="divider-container"
  >
    <View
      className="flex-row items-center w-full gap-[5px]"
    >
      <View
        aria-orientation="horizontal"
        className="w-full based-styles h-[1px] flex-1 bg-typography-secondary"
        testID="divider-with-title"
      />
      <Text
        className="based-styles font-regular text-[12px] tracking-[0px] text-typography-secondary -mt-[2px]"
      >
        Or login with
      </Text>
      <View
        aria-orientation="horizontal"
        className="w-full based-styles h-[1px] flex-1 bg-typography-secondary"
        testID="divider-with-title"
      />
    </View>
  </View>
</View>
`;
